### ISSUE TITLE
add current volume setting as new preset 

### ISSUE THREAD

**Issue by alshakhasm** [Slicer/Slicer#5505]
## Is your feature request related to a problem? Please describe.

I always start creating my volume rendering from scratch, however, that takes a long time and a lot of time and guessing 

## Describe the solution you'd like

creat a simple " create a custom preset " in the volume rendering module

**Comment by lassoan** [comment]
Add (+) and remove (-) buttons could be added right next to the preset selector to save or remove the current volume rendering preset.

Image could be generated by taking a screenshot of the first 3D view where the current preset is used in.

Presets could be stored in folder of Slicer.ini file, in a VolumeRenderingPresets subfolder.

**Comment by chir-set** [comment]
Following https://discourse.slicer.org/t/setting-the-volume-property-to-be-used-when-a-volume-is-rendered-automatically/16653/5 :

I don't find any voting facility here or I'm missing it.

Just adding that it would be great feature. I too have a custom volume property file (*.vp) that I load manually whenever I need it. In brief, the volume property file sort fo 'removes' calcifications on CT angioscans.

+1

**Comment by muratmaga** [comment]
> I don't find any voting facility here or I'm missing it.

Use the smiley button at the top of the issue (on the right) and give select thumbs up. THere are already two votes for it....
But I think we need something bit more robust because you can technically add smileys to each individual reply (e.g., this).

**Comment by lassoan** [comment]
> I think we need something bit more robust because you can technically add smileys to each individual reply (e.g., this).

I agree, but there is no better option right now. See details here: https://discourse.slicer.org/t/how-to-prioritize-feature-requests/16360/5?u=lassoan

**Comment by muratmaga** [comment]
@lassoan @pieper Given that soon we will be working on Slicer 6.0, I would like this long standing issue(s) of users not being able to add their custom setting (volume rendering and color tables specifically) in a persistent manner solved, along with some minor usability improvement. 

Should we continue on this topic here, or open a new issue?

**Comment by pieper** [comment]
> Should we continue on this topic here, or open a new issue?

Probably this issue is fine.  The discussion here makes sense, someone just needs to spend the time to act on it.

**Comment by muratmaga** [comment]
There are few related things, so I wasn't sure:
1. Making VP nodes exposed in SH, so that operations like cloning is possible (this would make iterative experimentation for the transfer function much more simple, without having to worry about recovering a state. I know the scene views does something similar so perhaps we can make that more convenient). 
2. Updating VP format something more meaningful (json, csv?)
3. The ability store custom settings (like color tables) persistently (not just per slicer installation, someplace like ~/.config/slicer/custom-settings). 

If we want we can split them up into different topics, but to me they are conceptually related.

**Comment by lassoan** [comment]
> Updating VP format something more meaningful (json, csv?)

@Sunderlandkyl has already implemented this (it was necessary because we needed specification of volume rendering of multi-channel images for SlicerHeart, for B-mode + Doppler echo images).

> Making VP nodes exposed in SH, so that operations like cloning is possible

This would be easy to implement (the volume rendering subject hierarchy plugin could claim the volume property nodes). However, it would not be consistent that the volume property node showed up in the SH tree, but other display and display-like nodes, such as color tables do not show up.

I'm wondering if a "preset manager" for volume rendering presets and color tables would be a good solution. See for example the preset manager (for registration parameter settings) in Elastix extension:

<img width="1469" height="962" alt="Image" src="https://github.com/user-attachments/assets/4236339b-ea21-42af-bfc4-287a644ae103" />

**Comment by muratmaga** [comment]
> However, it would not be consistent that the volume property node showed up in the SH tree,

Actually why is SH expected to work that way? I sure don't want (or expect) the SH to display all the built-in color tables or the VP presets, but if I end up creating my own color table, or my VP preset during that scene activity, I think it make sense that they show up in the SH much like a my new ROI shows up. Granted I don't have the technical depth to understand the issue, but from a user point of view, this is perfectly fine (I see no inconsistency there). 

> I'm wondering if a "preset manager" for volume rendering presets and color tables would be a good solution.

Possibly, but the devil is in the details. That menu seems very complex and intimidating, and if I am trying to adjust a new custom preset, it doesn't help so much to display the contents of the file, but I would rather see what the settings does visually. 

Now if I the SH functionality I described above is possible, then we could add a SH function like "Add to Preset Manager" or something along the line to make them persistent. I might create 5-10 different custom VP (or color table) to test out things in that scene, but perhaps maybe only one is going to be what I consider worthwhile to preserve.

**Comment by cpinter** [comment]
> it would not be consistent that the volume property node showed up in the SH tree, but other display and display-like nodes, such as color tables do not show up.
I'm wondering if a "preset manager" for volume rendering presets and color tables would be a good solution. See for example the preset manager (for registration parameter settings) in Elastix extension

I vote for the preset manager. First of all, display nodes do not show up in SH (they are not "data nodes", and I think it would be unexpected if we started to add display nodes). Second, the volume rendering presets are stored in a private scene, so it would be also confusing if the user saw a few volume property nodes but not all of them.

> I think it make sense that they show up in the SH much like a my new ROI shows up

I don't agree. ROI clearly contains data specified by the user. Display nodes are accessories that belong to data nodes, specifying how that particular data is shown in certain views. Showing display nodes would be also confusing because of this complexity (any data node may have multiple display nodes, one for each view for example), and users are not familiar with the internals to this degree so they would not understand what is happening with all the display nodes.

In summary, I have a strong preference not to include display nodes in SH, and instead add an explicit widget for handling volume properties.

The Volume Rendering module is bleeding from some wounds anyway. The main one I think has been addressed (which was the combobox with the separate eye icon - now we have SH but the eye icon in the tree also works for 2D while the other one only in 3D). However, there are still some problems, including the erratic shift slider with unexpected range and sensitivity, the unintuitive transfer function editing, the general high complexity, etc.

I think we either do someting small, undisruptive, and additive (like the + / - buttons to add/remove preset and the manager dialog), or a complete revamp (for which we should probably have a longer breakout at the next PW).

**Comment by muratmaga** [comment]
> I vote for the preset manager. First of all, display nodes do not show up in SH (they are not "data nodes", and I think it would be unexpected if we started to add display nodes). Second, the volume rendering presets are stored in a private scene, so it would be also confusing if the user saw a few volume property nodes but not all of them.

I think we are arguing from different points of view. If we ask normal (non-developer) Slicer users "what shows up in SH module", I don't think that's the answer you will get would be what you describe. It will be more like "somethings show and some dont", or maybe along the lines of "common things you do in Slicer". That argument is all about engineering, not from usability perspective. But from what I am  hearing this is probably something involved to implement anyways (which I suspect we won't have the funds), so I will not press the issue much further, beyond saying we have to continue supporting SH to be the `command center`of Slicer. I think I have been using Slicer 5-6 years until I discovered there is actually a `Colors` module and that the color tables are editable. 

> I vote for the preset manager.

I don't object to this, what I am saying it needs to be easy to use and discoverable. That screenshot above is not something I consider either. 

> I think we either do someting small, undisruptive, and additive (like the + / - buttons to add/remove preset and the manager dialog), or a complete revamp (for which we should probably have a longer breakout at the next PW).

I completely agree about revamping the VR module. However, I am not sure there will be funding for it in near future. I tried twice getting support (one with Kitware much bigger) and one with Queens focused on redesigned without touching the code much, and neither was funded. I don't see that pattern changing near future. Though happy to help whatever I can, if there is any prospect of funding elsewhere.

**Comment by muratmaga** [comment]
Any other thoughts on how to improve Preset Manager?
